sherpa/models/model.py

```diff
@@ -731,6 +731,75 @@ def __setattr__(self, name: str, val) -> None:
         for alias in val.aliases:
             self._par_index[alias] = val
 
+    def get_parts(self,
+                  include_composites : bool = True,
+                  remove_duplicates : bool = False) -> Sequence[Model]:
+        """Return the parts of the model.
+
+        Parameters
+        ----------
+        include_composites : bool, optional
+            If True, include the components of any composite models.
+            If False, only include the top-level components.
+        remove_duplicates : bool, optional
+            If True, remove any duplicate components. This is
+            useful if the same component is used in multiple
+            places in the model.
+
+        Returns
+        -------
+        modellist : list
+            The parts of the model.
+        """
+        # Note:
+        # include_composites and remove_duplicates is not used in the base class,
+        # but it is used in the CompositeModel class.
+        return [self]
+
+    def get_components_by_name(self, name: str) -> list[Model]:
+        """Return the components of the model with the given name.
+
+        Parameters
+        ----------
+        name : str
+            The name of the component to return.
+
+        Returns
+        -------
+        modellist : list
+            The components of the model with the given name.
+        """
+        parts = [p for p in self.get_parts(remove_duplicates=True) if p.name == name]
+        if len(parts) == 0:
+            raise KeyError(f"No components found with name '{name}'")
+        return parts
+
+    def get_components_by_class(self, cls: type, subclass_ok: bool = True) -> list[Model]:
+        """Return the components of the model with the given type.
+
+        Parameters
+        ----------
+        type : type
+            The type of the component to return.
+        subclass_ok : bool, optional
+            If True, return components that are subclasses of the given
+            type. If False, only return components that are exactly of
+            the given type.
+
+        Returns
+        -------
+        list[Model]
```

Comment:
On line 769 we have "modellist : list" and here just "list[Model]". We should gave the same in both.

Reply:
done.